api/v1alpha1/base_types.go

```diff
@@ -279,6 +279,14 @@ type RestoreSource struct {
 	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
 	TargetRecoveryTime *metav1.Time `json:"targetRecoveryTime,omitempty" webhook:"inmutable"`
+	// FileName is the file within the source to be restored.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	FileName *string `json:"fileName,omitempty" webhook:"inmutableinit"`
+	// Type is the type of backup. (mariabackup or mariadump)
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	Type *string `json:"type,omitempty" webhook:"inmutableinit"`
```

Comment:
Maybe we can leave as `string` and default it here?
https://github.com/mariadb-operator/mariadb-operator/blob/0179c43be88b483168cee3d54939d743d8dabcc5/api/v1alpha1/base_types.go#L295

This would reduce the risk of nil pointers, which by the way I noticed we are getting some in the CI tests:
- https://github.com/mariadb-operator/mariadb-operator/actions/runs/7332592001/job/19966968808?pr=273

In this line specifically:
https://github.com/mariadb-operator/mariadb-operator/pull/273/files#diff-23dfecbfdd15c2c0a28d94dfb1bae6d933f91a37124325caf0257ce5aa9cfcafR106

Also, what about creating an enum and document the values like we do here:
https://github.com/mariadb-operator/mariadb-operator/blob/0179c43be88b483168cee3d54939d743d8dabcc5/api/v1alpha1/mariadb_galera_types.go#L17